components/DataTable/src/DataTable/DataTable.cs

```diff
@@ -93,7 +93,12 @@ protected override Size MeasureOverride(Size availableSize)
                 // then invalidate the child arranges [don't re-measure and cause loop]...)
 
                 // For now, we'll just use the header content as a guideline to see if things work.
-                column.Measure(new Size(availableSize.Width - fixedWidth - autoSized, availableSize.Height));
+
+                // Avoid negative values when columns don't fit `availableSize`. Otherwise the `Size` constructor will throw.
+                double width = availableSize.Width - fixedWidth - autoSized;
+                if (width < 0)
+                    width = 0;
```

Comment:
```suggestion
                width = Math.Max(width, 0);
```

Reply:
Thanks for the review @xperiandri! Looks better this way, just updated the PR with the changes you suggested.